chacha20_poly1305/src/chacha20.rs

```diff
@@ -52,12 +52,10 @@ impl Nonce {
 ///    * For-each loops are easy for the compiler to recognize as vectorizable.
 ///    * The type is a based on an array instead of tuple since the heterogeneous
 ///      nature of tuples can confuse the compiler into thinking it is not vectorizable.
-///    * Memory alignment lines up with SIMD size.
 ///
 /// In the future, a "blacklist" for the alignment option might be useful to
 /// disable it on architectures which definitely do not support SIMD in order to avoid
 /// needless memory inefficientcies.
-#[repr(align(16))]
```

Comment:
If it's only on recent version then perhaps conditionally compile it with alignment?

Reply:
I'll run some tests to see if it will be worth it

Reply:
I ran a small matrix of tests involving different permutations of 1.63 compiler, the bleading edge compiler, and alignment sizes. Didn't see anything very noticeable in the results. An `align(16)` might help a bit for the 1.63 compiler, like 1%, but may just be noise. With that said, I plan to do a deeper dive on this with the cpu-specific tests as well. But for this patch the complexity probably isn't worth it.